### PR TITLE
Source Manager: Store stats of duplicates and remember their download.

### PR DESCRIPTION
**Pull request by s-l-teichmann**
Solve #491

### CODE REVIEW

**Review by koplas — APPROVED** [review]
LGTM

Update:
I tested it with a source that already contains imported documents. I get the following error in the feed log:
`storing "https://security.access.redhat.com/data/csaf/v2/advisories/2024/rhsa-2024_4484.json" failed: already in database ERROR: current transaction is aborted, commands ignored until end of transaction block (SQLSTATE 25P02)`.

The downloads table does not contains a single entry where `duplicate_failed` equals true.

**Comment by s-l-teichmann** [comment]
> LGTM
> 
> Update: I tested it with a source that already contains imported documents. I get the following error in the feed log: `storing "https://security.access.redhat.com/data/csaf/v2/advisories/2024/rhsa-2024_4484.json" failed: already in database ERROR: current transaction is aborted, commands ignored until end of transaction block (SQLSTATE 25P02)`.
> 
> The downloads table does not contains a single entry where `duplicate_failed` equals true.

Argh! Forgot to commit the transaction. Should be fixed now.

File: pkg/models/import.go [hunk 1]
```diff
@@ -359,6 +359,9 @@ func ImportDocumentData(
 				if err2 := inTx(ctx, tx, 0, true); err2 != nil {
 					return 0, errors.Join(ErrAlreadyInDatabase, err2)
 				}
```
Comment (koplas) [inline pkg/models/import.go:361]: Does this statement need to be removed? I still get the same errors.

Comment (s-l-teichmann) [reply pkg/models/import.go]: No, the error canceled the whole transaction so it was not possible to write anything into the transaction afterwards. This is now circumvented by isolating the storing of the document with a save point.
This results in only rolling back to this and keep the transaction intact for further writes.

This should work now.

**Comment by koplas** [comment]
It now works.

**Pull request merged**